### DISCUSSION TITLE
Keys are not distributing evenly across nodes.

### DISCUSSION THREAD

**Discussion by cmreddyv** [redis/go-redis#2634] [Q&A]
Seeing weird behaviour .. one server is at 24k keys while others are at 3 digit numbers... the key distribution is not happening properly. how to solve this and get the keys evenly distributed.

Redis Server version: 6.2.7

GoRedis client version: v7

Tried with Redis Server version: 7.0.8 and Client version v9( same behaviout)

<img width="935" alt="image" src="https://user-images.githubusercontent.com/33657970/232854220-248daa25-ff5f-44fd-8156-3d75f8947ba1.png">

**Comment by SoulPancake** [comment]
@cmreddyv Don't think it is related to the client.
Though If you have control over your key names, you can use hash tags to ensure a more even distribution of keys across your Redis nodes if that makes sense in your case

**Comment by cmreddyv** [comment]
@SoulPancake , not able to figure it out whether it is clientissue or server issue ... it is pushing the keys and also establishing connections to only one node in the Redis OSS server, which is making that particular server overloaded and causing time out issues.

Tried to upgrade the client to Redis V9 and server to Redis Server 7.0.8, but still the same.


`func NewRedisClusterClient(config RedisConfig) *redis.ClusterClient {
	redisClient := redis.NewClusterClient(&redis.ClusterOptions{
		Addrs:          config.Hosts,
		Password:       config.Password,
		DialTimeout:    10 * time.Second, // increase the timeout as needed
		ReadTimeout:    1 * time.Second,
		WriteTimeout:   1 * time.Second,
		PoolSize:       500,
		MinIdleConns:   10,
		PoolTimeout:    30 * time.Second,
		ReadOnly:       true,
		RouteByLatency: true,
	})`
	
	
	
<img width="940" alt="image" src="https://user-images.githubusercontent.com/33657970/233123234-839e0f3c-5447-4a2f-919b-14991ddf3ee6.png">

**Comment by Odysseusailoon** [comment]
Hi it's not a client side issue, the keys' dirstribution are depending on crc16, the client just send them to the corresponding slot, you can refer to https://redis.com/blog/redis-clustering-best-practices-with-keys/

**Comment by elena-kolevska** [comment]
@cmreddyv A very common cause of this problem is the misusage of [hashtags](https://redis.io/docs/reference/cluster-spec/#hash-tags). Do you know if you're using them for naming keys? If yes, you should consider the design of the hashtag. You should include some variable part inside of the curly braces.
